channeldb/invoices.go

```diff
@@ -1969,13 +1969,8 @@ func (d *DB) cancelHTLCs(invoices kvdb.RwBucket, invoiceNum []byte,
 		return nil, errors.New("cancel action on non-existent htlc(s)")
 	}
 
-	// Reserialize and update invoice.
-	var buf bytes.Buffer
-	if err := serializeInvoice(&buf, invoice); err != nil {
-		return nil, err
-	}
-
-	if err := invoices.Put(invoiceNum, buf.Bytes()); err != nil {
+	err := d.serializeAndStoreInvoice(invoices, invoiceNum, invoice)
```

Comment:
non-blocking: are there any reasons that we call this `serializeAndStoreInvoice` at multiple places? Instead of using it in `cancelHTLCs`, `addHTLCs` and `settleHodlInvoice` separately, I think we could just call it outside the `switch update.UpdateType {`, and return `d.serializeAndStoreInvoice`, seems like a few more lines were saved and fewer params used.

Or, do you plan to further make these actions into independent db tx(which is nice IMO)?

Reply:
Each one will be its own db interface method